### PR TITLE
some labels and help

### LINKED ISSUE #6: labels and help
Following the comments of @bgruening in #5 

labels and help is missing for all parameters

label and help could be added and the overall tool help could be improved.

Its not very useful to explain the parameters, for that you have label and help on the params. Instead write what the tool is doing, explain the output etc ...

### PR DESCRIPTION
**Pull request by volodymyrss**

### CODE REVIEW

**Comment by bgruening** [comment]
Yes, we do have some clustering, prediction, scikit-learn etc tools. Lets get those initial set in EU and then I can show you more tools.

>ables into CSV 

A short note, Galaxy has a LOT of tools that can work with the `tabular` datatype (tsv, tab separated) so if possible its recommended to stay with tabular and not csv.

File: tools/astropytools/fits2csv.xml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+        </configfile>
+    </configfiles>
+
+    <inputs>
+        <param type="data" name="input_fits" format="fits" label="FITS file to dump"/>
+        <param type="integer" name="hdu"  value="1" min="1" label="Select input HDU number"/>
+        <param type="select" name="output_delimiter" value="," label="Output delimiter">
+            <option value="," selected="true">"," (coma)</option>
+            <option value=" ">" " (space)</option>
+            <option value="\t">"\t" (tab)</option>
```
Comment (bgruening) [inline tools/astropytools/fits2csv.xml:33]: make that default?

File: tools/astropytools/fits2csv.xml [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+    </outputs>
+    <tests>
+        <test>
+            <param name="input_fits" value="WFPC2u5780205r_c0fx.fits"/>
+            <param name="hdu" value="1"/>
+            <output name="output_csv" file="fitsinfo.csv"/>
+        </test>
+    </tests>
+    <help><![CDATA[
+Extract CSV table from FITS file HDU Table. Resulting CSV file can be used by many existing Galaxy tools and visualisation plugins.
```
Comment (bgruening) [inline tools/astropytools/fits2csv.xml:47]: Not only CSV this tool can supports more ...

**Review by bgruening — APPROVED** [review]

**Comment by volodymyrss** [comment]
> Not sure why the tests are failing, does it fail locally as well?

They are not failing locally. This is weird.

File: tools/astropytools/fits2table.xml [hunk 3]
```diff
@@ -20,31 +20,31 @@ with fits.open("$input_fits", memmap=True) as hdu_list:
...
-            <option value="," selected="true">"," (coma)</option>
+            <option value="\t" selected="true">"\t" (tabular)</option>
+            <option value=",">"," (coma)</option>
             <option value=" ">" " (space)</option>
-            <option value="\t">"\t" (tab)</option>
         </param>
     </inputs>
     <outputs>
-        <data name="output_csv" format="csv"/>
+        <data name="output" format="csv,ssv,tabular"/>
```
Comment (bgruening) [inline tools/astropytools/fits2table.xml:37]: outputs can only have ONE format ... you need to use this here to change the format depending on the user selection: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-outputs-data-change-format-when

Comment (bgruening) [reply tools/astropytools/fits2table.xml:37]: And I see now ... Galaxy does not support SSV yet, never heard about it until now :)

Comment (volodymyrss) [reply tools/astropytools/fits2table.xml:37]: > outputs can only have ONE format ... you need to use this here to change the format depending on the user selection: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-outputs-data-change-format-when

I was experimenting. Linting did not complain. I guess it just sees it as long type with comas?

Comment (bgruening) [reply tools/astropytools/fits2table.xml:37]: Mh ... do you know if XSD can check for those things?

Comment (volodymyrss) [reply tools/astropytools/fits2table.xml:37]: Maybe, I did not use it, but it seems? https://www.w3schools.com/XML/schema_facets.asp https://stackoverflow.com/questions/8925706/xml-schema-how-to-restrict-attribute-by-enumeration ?

Comment (volodymyrss) [reply tools/astropytools/fits2table.xml:37]: It causes warning in the test though but not an error.

**Comment by volodymyrss** [comment]
> A short note, Galaxy has a LOT of tools that can work with the `tabular` datatype (tsv, tab separated) so if possible its recommended to stay with tabular and not csv.

I saw some examples with CSV so used it, it's also more typical in our area, though it's not a big difference since binary table of its is better for us in most cases.

By the way, is there favorable image format in Galaxy, so we can convert to there and benefit from Galaxy tools?

**Comment by bgruening** [comment]
No preferred image format afaik. We do have graphicsmagic to convert ect... So that should be fine for the first version.

**Comment by volodymyrss** [comment]
> No preferred image format afaik. We do have graphicsmagic to convert ect... So that should be fine for the first version.

Ok, image in the sense of multi-dimensional maps I mean. Lot's of analysis can be done on it. But it can be done on `png` as well, even with opencv.
 
I suppose the focus of this version is still to integrate astronomical data into galaxy, allowing to exploit the synergy and

**Comment by bgruening** [comment]
I know communities that used tiff in various forms, but we are shifting to zarr (ome-zarr) atm.

File: tools/astropytools/fits2csv.xml (hunk 2 above)
Comment (volodymyrss) [reply tools/astropytools/fits2csv.xml]: I changed the text.

**Comment by volodymyrss** [comment]
> 

thanks, I make an issue #9 for the future

**Review by bgruening — APPROVED** [review]
Cool ship it!

**Comment by volodymyrss** [comment]
Ok, [done](https://toolshed.g2.bx.psu.edu/repository/browse_repositories_in_category?id=292389a45f1a238a&message=&status=done). Is it possible to make them available in usegalaxy.eu at some point?

**Comment by bgruening** [comment]
Yes, it is please follow step 3 https://galaxyproject.org/news/2020-08-22-three-steps-to-galaxify-your-tool/ :)

**Comment by volodymyrss** [comment]
> Yes, it is please follow step 3 https://galaxyproject.org/news/2020-08-22-three-steps-to-galaxify-your-tool/ :)

ok, [here](https://github.com/usegalaxy-eu/usegalaxy-eu-tools/pull/594)

**Pull request merged**